### PR TITLE
feature: add support for test specs names

### LINKED ISSUE #1849: Enable users to put a description into a test spec, allowing it to document it
**Describe the enhancement you'd like to see**

Sometimes a spec assertion could be difficult to understand at first glance. We could add a description field on a spec to allow the user to explain what is that assertion.

For instance, I have these specs:
image

And I wanted to do those explanations:
image

### PR DESCRIPTION
**Pull request by jorgeepc**
This PR enables users to add a `name`/`description` to a test spec.

## Changes

- Add support for the `name` field in test specs

## Fixes

- fixes #1849 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Loom CLI

https://www.loom.com/share/23b42ec43004483da2ed9221bb56d283

## Loom UI

https://www.loom.com/share/9eb56b92a1904a1a893d163874c5af1b

## Screenshots CLI

Screenshot 2023-02-02 at 13 21 58
Screenshot 2023-02-02 at 13 22 24
Screenshot 2023-02-02 at 15 30 09

### CODE REVIEW

File: web/src/components/TestSpec/Header.tsx [hunk 1]
```diff
@@ -7,19 +7,24 @@ interface IProps {
...
   return (
     <S.Column>
-      <Editor
-        type={SupportedEditors.Selector}
-        editable={false}
-        basicSetup={{lineNumbers: false}}
-        placeholder="Selecting All Spans"
-        value={title || 'All Spans'}
-      />
+      {title ? (
```
Comment (xoscar) [inline web/src/components/TestSpec/Header.tsx:17]: I'm wondering if we want to show the selector on hover or something like that, should it be only displayed after clicking the test spec now?

File: web/src/components/TestSpec/TestSpec.tsx [hunk 2]
```diff
@@ -43,7 +44,8 @@ const TestSpec = ({
         affectedSpans={spanIds.length}
         assertionsFailed={totalPassedChecks?.false ?? 0}
         assertionsPassed={totalPassedChecks?.true ?? 0}
-        title={selector}
+        selector={selector}
+        title={!selector && !name ? 'All Spans' : name}
```
Comment (xoscar) [inline web/src/components/TestSpec/TestSpec.tsx:48]: Should we calculate this from the model and then use `name` everywhere instead?

File: web/src/components/TestSpecDetail/Header.tsx [hunk 3]
```diff
@@ -69,13 +71,17 @@ const Header = ({
...
       <Divider />
 
-      <Editor
-        type={SupportedEditors.Selector}
-        editable={false}
-        basicSetup={{lineNumbers: false}}
-        placeholder="Selecting All Spans"
-        value={title || 'All Spans'}
-      />
+      {title && <S.Title>{title}</S.Title>}
```
Comment (xoscar) [inline web/src/components/TestSpecDetail/Header.tsx:74]: In the video, there seems to be a weird misalignment between these two. Could we also increase either the font-weight or size so the title looks more prominent?

**Review by xoscar — APPROVED** [review]
Great job my dude!

**Review by mathnogueira — APPROVED** [review]
CLI code is good for me!

**Pull request merged**